src/Components/Tasks.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useState } from "react";
+import Task from "./Task";
+import Swal from "sweetalert2";
+import Modal from "./Modal";
+
+export default function Tasks() {
+  const [dataParsed, setDataParsed] = useState([]);
+  const [toggle, setToggle] = useState(0);
+  const [token, setToken] = useState("");
+
+  console.log(token);
+
+  function showSuccessPopup()
+  {
+    Swal.fire("Good job!", "You clicked the button!", "success");
+  }
+
+
+  function getDataFromLocalStorage() {
+    const jsonFormatData = localStorage.getItem("Tasks");
+    try {
+      return JSON.parse(jsonFormatData) || [];
+    } catch {
+      return [];
+    }
+  }
+
+  function addToLocalStorage(name, assignee) {
+    let dataObject = {
+      id: `${Math.random() * 1232}`,
+      done: 0,
+      name,
+      assignee,
+      
+    };
+    dataParsed.push(dataObject);
+    setDataParsed(dataParsed);
+    let localStorageData = JSON.stringify(dataParsed);
+    localStorage.setItem("Tasks", localStorageData);
+    showSuccessPopup();
+    setTimeout(() => {
+      window.location.reload();
+    }, 1400);
+  }
+
+  function displayData() {
+    return dataParsed
+      .filter((task) => task.name.toLowerCase().includes(token))
+      .map((element, key) => {
+        return element.done == toggle ? (
+          <Task
+            id={key}
+            name={element.name}
+            assignee={element.assignee}
+            doneAttribute={element.done}
+            setItemDone={() => setItemDone(key)}
+            deleteTask={() => deleteTask(key)}
+          />
+        ) : (
+          ""
+        );
+      });
+  }
+
+  function setItemDone(id) {
+    dataParsed.find((element) => {
+      if (dataParsed.indexOf(element) == id) {
+        dataParsed[id].done = 1;
+      }
+      setDataParsed(dataParsed);
+      const localStorageData = JSON.stringify(dataParsed);
```

Comment:
you are doing this 3 times in this file, isn't this an indicator for a new function??